Environment/CTP_environment.py

```diff
@@ -94,28 +119,67 @@ def step(self, state:EnvState,action:jnp.array) -> tuple[Observation,EnvState,in
+                    ),
+                    True,
+                )
+            ),
+            lambda node1, node2: jnp.argmax(
+                jnp.append(
+                    jnp.logical_and(
+                        jnp.logical_and(
+                            self.true_graph.senders == node2,
+                            self.true_graph.receivers == node1,
+                        ),
+                        self.true_graph.edges["blocked_status"] == 0,
+                    ),
+                    True,
+                )
+            ),
+            node1,
+            node2,
+        )
         return index_of_edge
 
-    #This is currently not jax-compatible because edge_indices size varies
-    def get_obs(self, state:EnvState) -> Observation:
+    # This is currently not jax-compatible because edge_indices size varies
+    def get_obs(self, state: EnvState) -> Observation:
         # Get the blocking status of the stochastic edges connected to a node
-        for agent_num in range(self.num_agents):                
-            edge_indices = jnp.where(jnp.logical_and(jnp.logical_or(self.true_graph.senders == state.agents_pos[agent_num],self.true_graph.receivers == state.agents_pos[agent_num]), self.true_graph.edges['blocked_prob'] > 0))
-            observation = FrozenDict({
-                agent_num: FrozenDict({
-                    (int(self.true_graph.senders[edge_indices[0][i]]), int(self.true_graph.receivers[edge_indices[0][i]])): 
-                    int(self.true_graph.edges['blocked_status'][edge_indices[0][i]])  # Converting the edge status to int as well
-                    for i in range(len(edge_indices[0]))
-                })
-            })
+        for agent_num in range(self.num_agents):
+            edge_indices = jnp.where(
+                jnp.logical_and(
+                    jnp.logical_or(
+                        self.true_graph.senders == state.agents_pos[agent_num],
+                        self.true_graph.receivers == state.agents_pos[agent_num],
+                    ),
+                    self.true_graph.edges["blocked_prob"] > 0,
+                )
+            )
+            observation = FrozenDict(
+                {
+                    agent_num: FrozenDict(
+                        {
+                            (
+                                int(self.true_graph.senders[edge_indices[0][i]]),
+                                int(self.true_graph.receivers[edge_indices[0][i]]),
+                            ): int(
+                                self.true_graph.edges["blocked_status"][
+                                    edge_indices[0][i]
+                                ]
+                            )  # Converting the edge status to int as well
+                            for i in range(len(edge_indices[0]))
+                        }
+                    )
+                }
+            )
         return observation
-                
```

Comment:
The environment is definitely something that should be unit tested. Make some small maps and check that the actions do what you expect them to do, and return the correct reward and observations